src/dsl/cb.rs

```diff
@@ -148,13 +194,24 @@ pub fn select<
     let when_true = when_true.into();
     let when_false = when_false.into();
 
+    match selector.typing {
+        Typing::Boolean | Typing::Unknown => (),
+        _ => panic!("Expected Boolean or Unknown selector, got AntiBooly (selector: {})", selector.annotation)
+    }
+
+    let typing: Typing = match (when_true.typing.clone(), when_false.typing) {
+        (Typing::AntiBooly, Typing::AntiBooly) | (Typing::Boolean, Typing::Boolean) => when_true.typing,
```

Comment:
this is more like:

typing = if when_true.typing == when_false.typing then when_true.typing else Unknown.

You don't need to say `typing: Typing` I think the compiler will guess it

Reply:
Fixed!

Reply:
Also simplified similar syntax else where.